### PR TITLE
quick patch on nvfuser getnv/reshape to handle sequence with NumberProxy

### PR DESCRIPTION
**Pull request by jjsjann123**
1. recursively go through `getnv` on sequence to retrieve dynamic scalar for nvfuser.
2. update reshape parsing in nvfuser executor to allow NumberProxy in new shape.

### CODE REVIEW

**Comment by jjsjann123** [comment]
PR not functional yet, blocked by https://github.com/NVIDIA/Fuser/pull/2530

File: thunder/tests/test_jit_general.py [hunk 1]
```diff
@@ -1090,3 +1090,32 @@ def forward(self, x):
...
+    ("cpu", "cuda"),
+)
+def test_cache_symbolic_values_reshape(device):
+    if not torch.cuda.is_available():
+        pytest.skip("CUDA not available")
+
+    a = torch.randn((4, 8, 6), device=device)
+
+    def foo(t, batch_size):
+        return t.reshape(batch_size, -1).sum(-1)
```
Comment (jjsjann123) [inline thunder/tests/test_jit_general.py:1106]: we might want to discuss later on how we would like to represent dynamic shape. (design review due)

Meanwhile, this is a quick patch coming from patching @kiya00 's issue: https://github.com/Lightning-AI/lightning-thunder/pull/451#issuecomment-2189428679

File: thunder/executors/nvfuserex_impl.py [hunk 2]
```diff
@@ -1198,11 +1200,12 @@ def _reshape_check(a: TensorProxy, shape: list[int]) -> bool:
 
 def reshape(a: TensorProxy, shape: list[int], *, fd: FusionDefinition, lc_to_nv_map: dict) -> Any:
```
Comment (IvanYashchuk) [inline thunder/executors/nvfuserex_impl.py:1201]: We don't do any type checking, but what would be a more accurate type hint for `shape`?

**Review by IvanYashchuk — APPROVED** [review]

File: thunder/tests/test_jit_general.py (hunk 1 above)
Comment (kiya00) [reply thunder/tests/test_jit_general.py:1106]: Thanks for the fix, I'll try it once it merged

**Review by kiya00 — APPROVED** [review]

File: thunder/executors/nvfuserex_impl.py (hunk 2 above)
Comment (jjsjann123) [reply thunder/executors/nvfuserex_impl.py:1201]: shape is going to be a `list[int, NumberProxy]` moving forward. We'll finalize that in a design review for dynamic shape :crossed_fingers:

**Review by t-vi — APPROVED** [review]
Thank you @jjsjann123 @kiya00 @IvanYashchuk

**Pull request merged**